pkg/yqlib/operator_divide_test.go

```diff
@@ -0,0 +1,121 @@
+package yqlib
+
+import (
+	"testing"
+)
+
+var divideOperatorScenarios = []expressionScenario{
+	{
+		skipDoc:    true,
+		document:   `[{a: foo_bar, b: _}, {a: 4, b: 2}]`,
+		expression: ".[] | .a / .b",
+		expected: []string{
+			"D0, P[0 a], (!!seq)::- foo\n- bar\n",
+			"D0, P[1 a], (!!float)::2\n",
+		},
+	},
+	{
+		skipDoc:    true,
+		document:   `{}`,
+		expression: "(.a / .b) as $x | .",
+		expected: []string{
+			"D0, P[], (doc)::{}\n",
+		},
+	},
+	{
+		description: "String split",
+		document:    `{a: cat_meow, b: _}`,
+		expression:  `.c = .a / .b`,
+		expected: []string{
+			"D0, P[], (doc)::{a: cat_meow, b: _, c: [cat, meow]}\n",
+		},
+	},
+	{
+		description:    "Number division",
+		subdescription: "The result during divison is calculated as a float",
+		document:       `{a: 12, b: 2.5}`,
+		expression:     `.a = .a / .b`,
+		expected: []string{
+			"D0, P[], (doc)::{a: 4.8, b: 2.5}\n",
+		},
+	},
+	{
+		skipDoc:     true,
+		description: "Custom types: that are really strings",
+		document:    "a: !horse cat_meow\nb: !goat _",
+		expression:  `.a = .a / .b`,
+		expected: []string{
+			"D0, P[], (doc)::a: !horse\n    - cat\n    - meow\nb: !goat _\n",
```

Comment:
in this case I think it'd be more correct if `a` is an array, but each element is a `horse`

Reply:
The behavior came from string split which sorta seems like it doesn't support custom tags yet.  Should I try to fix it there?

```bash
❯ echo $'a: !horse cat_meow\nb: !goat _' | yq '.a | split("_")'
- cat
- meow
```

I also just found this interesting bug

```bash
❯ echo $'a: !horse cat_meow\nb: !goat _' | yq '.a | split(.b)'
- c
- a
- t
- _
- m
- e
- o
- w
```

Reply:
oh I see - yeah I think so 🤔 , that would be appreciated

Reply:
Actually, I'm not sure I can change it easily.  It seems the assignment operator won't change custom tags.  Changing that would be a breaking change.

```bash
❯ echo $'a: !horse cat_meow\nb: !goat _' | yq '.a = {} | .c = {} | .[] | tag'
!horse
!goat
!!map

❯ echo $'a: !horse cat_meow\nb: !goat _' | yq '.a = {} | .c = {} | .c = [] | .[] | tag'
!horse
!goat
!!seq
```

Reply:
alright - leave it, no worries